### PR TITLE
feat: SQS devices

### PR DESCRIPTION
**Pull request by Avivsalem**

### CODE REVIEW

**Comment by Avivsalem** [comment]
1. add tests (using moto)
2. look at some edge cases (look for # TODO: in the code
3. add documentation

File: messageflux/iodevices/sqs/sqs_input_device.py [hunk 1]
```diff
@@ -66,82 +67,62 @@ def __init__(
...
-
-        sqs_messages = self._queue.receive_messages(
-            MessageAttributeNames=["All"],
-            MaxNumberOfMessages=self._max_messages_per_request,
-        )
+        if timeout is None:
+            sqs_messages = self._queue.receive_messages(
+                MessageAttributeNames=self._included_message_attributes,
+                MaxNumberOfMessages=self._max_messages_per_request,
+            )  # TODO: what's the visibility timeout? should we extend it?
```
Comment (noamisr) [inline messageflux/iodevices/sqs/sqs_input_device.py:94]: The visibility timeout usually configured in the queue created and it is default to the queue.
Why do we want to extend is?

File: messageflux/iodevices/sqs/sqs_input_device.py [hunk 2]
```diff
@@ -150,8 +131,8 @@ class SQSInputDeviceManager(SQSManagerBase, InputDeviceManager[SQSInputDevice]):
     SQS input device manager
     """
 
-    def __init__(self):
-        super(SQSManagerBase, self).__init__()
+    def __init__(self, sqs_resource: 'SQSServiceResource'):
```
[the remark this answers is not in the document]
Comment (Avivsalem) [reply messageflux/iodevices/sqs/sqs_input_device.py:134]: there are ALOT of parameters that can be changed while creating the resource... i wanted to give the user the option to init it as he wants. (otherwise, you're losing degrees of freedon

[the remark this answers is not in the document]
Comment (noamisr) [reply messageflux/iodevices/sqs/sqs_input_device.py:134]: From the experience of working with the library, it is far less convenient to configure `boto3.resource` in a pythonic way.

In all of the use cases i've experienced (working locally with a temp aws profile and initializing the boto in a k8s pod) initializing boto without args and setting (many) combinations of aws environment variables gives you all the freedom you need as a developer.

If there is a param that boto does not look for in env var, and you think is important to the configuring, then mention it explicitly.

**Comment by Avivsalem** [comment]
maybe I can make it a default variable...

File: messageflux/iodevices/sqs/sqs_input_device.py (hunk 1 above)
Comment (noamisr) [reply messageflux/iodevices/sqs/sqs_input_device.py:94]: If, as the developer of the pipeline, the max time you give to process message is X, then you should configure X as the visibility timeout for the queue. 
IMO, this decision is out of the scope of the sqs input device implementation.

File: messageflux/iodevices/sqs/sqs_input_device.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Avivsalem) [reply messageflux/iodevices/sqs/sqs_input_device.py]: fixed it. now you can give it, or not(and it will create it from env vars)
